app/connection.go

```diff
@@ -4,67 +4,93 @@ import (
 	"context"
 	"crypto/tls"
 	"fmt"
-	"strings"
+	"net/url"
 
+	"github.com/temporalio/tcld/app/credentials/apikey"
+	"github.com/temporalio/tcld/app/credentials/oauth"
 	"github.com/urfave/cli/v2"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/metadata"
 )
 
+const (
+	VersionHeader = "tcld-version"
+	CommitHeader  = "tcld-commit"
+)
+
 func GetServerConnection(c *cli.Context, opts ...grpc.DialOption) (context.Context, *grpc.ClientConn, error) {
-	serviceName := c.String(ServiceNameFlagName)
-	apiKeyID := c.String(APIKeyIDFlagName)
-	apiSecretKey := c.String(APISecretKeyFlagName)
-	enableDebugLogs := c.Bool(EnableDebugLogsFlagName)
-	serverAddr := c.String(ServerFlagName)
-	var credentialOption grpc.DialOption
-	parts := strings.Split(serverAddr, ":")
-
-	if len(parts) != 2 {
-		return nil, nil, fmt.Errorf("unable to parse hostname: %s", serverAddr)
+	addr, err := url.Parse(c.String(ServerFlagName))
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to parse server address: %s", err)
 	}
 
-	hostname := parts[0]
-	switch hostname {
-	case "localhost":
-		credentialOption = grpc.WithTransportCredentials(insecure.NewCredentials())
-	default:
-		credentialOption = grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{
-			MinVersion: tls.VersionTLS12,
-			ServerName: hostname,
-		}))
+	defaultOpts, err := defaultDialOptions(c, addr)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate default dial options: %s", err)
 	}
+
 	conn, err := grpc.Dial(
-		serverAddr,
-		append(
-			opts,
-			credentialOption,
-			grpc.WithUnaryInterceptor(
-				getRequestSignatureInterceptor(
-					serviceName,
-					apiKeyID,
-					apiSecretKey,
-					enableDebugLogs,
-				),
-			),
-		)...,
+		addr.String(),
+		append(defaultOpts, opts...)...,
 	)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("failed to dial `%s`: %v", addr.String(), err)
 	}
-	tokens, err := loadLoginConfig(c)
+	ctx := context.Background()
+	ctx = metadata.AppendToOutgoingContext(ctx, VersionHeader, getVersion())
+	ctx = metadata.AppendToOutgoingContext(ctx, CommitHeader, Commit)
+
+	return ctx, conn, nil
+}
+
+func defaultDialOptions(c *cli.Context, addr *url.URL) ([]grpc.DialOption, error) {
+	var opts []grpc.DialOption
+
+	creds, err := newRPCCredential(c)
 	if err != nil {
-		return nil, nil, err
+		return []grpc.DialOption{}, nil
```

Comment:
is this going to be confusing behavior to the user? Why did we deviate from returning an error?

Reply:
Ah that is a mistake, it should be returning the error. Will fix.